internal/bioscfg/bioscfg.go

```diff
@@ -0,0 +1,113 @@
+package bioscfg
+
+import (
+	"context"
+
+	"github.com/metal-toolbox/bioscfg/internal/config"
+	"github.com/metal-toolbox/bioscfg/internal/store/fleetdb"
+	"github.com/metal-toolbox/ctrl"
+	rctypes "github.com/metal-toolbox/rivets/condition"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	pkgName = "internal/bioscfg"
+)
+
+// BiosCfg BiosCfg Controller Struct
+type BiosCfg struct {
+	cfg     *config.Configuration
+	logger  *logrus.Entry
+	ctx     context.Context
+	fleetdb *fleetdb.Store
+	nc      *ctrl.NatsController
+}
+
+// New create a new BiosCfg Controller
+func New(ctx context.Context, cfg *config.Configuration, logger *logrus.Entry) (*BiosCfg, error) {
+	bc := &BiosCfg{
+		cfg:    cfg,
+		logger: logger,
+		ctx:    ctx,
```

Comment:
Contexts are not meant to be stored in a struct:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

https://pkg.go.dev/context

There's probably nothing bad that will happen if we do add it to a struct, but best to use them as intended.